internal/transport/vim.go

```diff
@@ -0,0 +1,252 @@
+package transport
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/myitcv/govim/internal/queue"
+	"gopkg.in/tomb.v2"
+	"io"
+	"strings"
+	"sync"
+	"time"
+)
+
+type vimTransport struct {
+	in  *json.Decoder
+	out *json.Encoder
+	log io.Writer
+
+	// callVimNextID represents the next ID to use in a call to the Vim
+	// channel handler. This then allows us to direct the response.
+	callVimNextID     int
+	callbackResps     map[int]Callback
+	callbackRespsLock sync.Mutex
+
+	eventQueue *queue.Queue
+	tomb       tomb.Tomb
+}
+
+func NewVimTransport(in io.Reader, out io.Writer, log io.Writer, eventQueue *queue.Queue) Transport {
+	return &vimTransport{
+		in:  json.NewDecoder(in),
+		out: json.NewEncoder(out),
+		log: log,
+
+		callVimNextID: 1,
+		callbackResps: make(map[int]Callback),
+
+		eventQueue: eventQueue,
```

Comment:
Replace this with a passed in callback that can enqueue things on the event queue, we really don't need the whole event queue